base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -261,7 +267,30 @@ impl LMDBDatabase {
                     height,
                     hash,
                     accumulated_difficulty,
+                    from,
                 } => {
+                    // for security we check that the best block does exist, and we check the previous value
+                    // we dont want to check this if the prev block has never been set, this means a empty hash of 32
+                    // bytes.
+                    if from != vec![0; 32] {
+                        let prev = fetch_best_block(&write_txn, &self.metadata_db)?;
+                        if from != prev {
+                            return Err(ChainStorageError::InvalidOperation(format!(
+                                "There was a change in best_block, the best block is suppose to be: ({}), but it \
+                                 currently is: ({})",
+                                from.to_hex(),
+                                prev.to_hex(),
+                            )));
+                        };
+                    }
+                    let header_height: Option<u64> = lmdb_get(&write_txn, &self.block_hashes_db, hash.as_slice())?;
```

Comment:
Use `lmdb_exists` (skips deserialisation of the value)
`if lmdb_exists(&write_txn, &self.block_hashes_db, hash.as_slice())? {`